### ISSUE TITLE
"Remove redundant parentheses" false positive: `(1 as u32) << 10`

### ISSUE THREAD

**Issue by benwh1** [rust-lang/rust-analyzer#21461]
**rust-analyzer version**: 0.3.2753-standalone

**rustc version**: rustc 1.92.0-nightly (4da69dfff 2025-10-01)

**editor or extension**: VSCode

**code snippet to reproduce**:
```rust
fn main() {
    _ = (1 as u32) << 10;
}
```

Rust-analyzer suggests the "remove redundant parentheses" action, but when they are removed, the code doesn't compile because `u32 <<` is parsed as the beginning of a type with generic parameters.
